alicloud/data_source_alicloud_rds_instances.go

```diff
@@ -0,0 +1,289 @@
+package alicloud
+
+import (
+	"log"
+	"regexp"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/rds"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAlicloudRdsInstances() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudRdsInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"db_instance_status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"db_instance_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```

Comment:
I think 'type' can replace 'db_instance_type'  to make it simple. And it should add validate, like Primary, Readonly, Guard, Temp and All

Reply:
Maybe `instance_type` is better here. We have others types like `charge_type` and `net_type`. Validation has been added here and for most of the attributes.